File: 0136-single-number/0136-single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        Map<Integer,Integer> map= new HashMap<>();
        for(int i: nums){
            map.put(i , map.getOrDefault(i,0)+1);
        }
        for(var x: map.entrySet()){
            if(x.getValue()==1){
                return x.getKey();
            }
        }
        return 0;
    }
}

File: 0136-single-number/0136-single-number.ts
// function singleNumber(nums: number[]): number {
    
// };


// /**
//  * @param {number[]} nums
//  * @return {number}
//  */
const singleNumber = (nums: number[]): number => {
    let ans:number = 0;
    for (let i :number = 0; i < nums.length; i++) {
        ans ^= nums[i];
    }
    return ans;
};
